github/repos.go

```diff
@@ -655,10 +655,11 @@ func (s *RepositoriesService) GetRequiredStatusChecks(ctx context.Context, owner
 	p := new(RequiredStatusChecks)
 	resp, err := s.client.Do(ctx, req, p)
 	if err != nil {
-		return nil, resp, err
+		// if it's just a 404, don't return that as an error
+		_, err = parseBoolResponse(err)
 	}
 
-	return p, resp, nil
+	return p, resp, err
```

Comment:
This last line can remain the success case:

```Go
return p, resp, nil
```

It'd also be consistent with https://github.com/google/go-github/blob/e8d46665e050742f457a58088b1e6b794b2ae966/github/activity_watching.go#L97-L103.